examples/demo-app/src/cloud-providers/dropbox/dropbox-provider.js

```diff
@@ -0,0 +1,475 @@
+    this._path = `/Apps/${window.decodeURIComponent(this.appName)}`;
+
+    // Initialize Dropbox API
+    this._initializeDropbox();
+  }
+
+  /**
+   * This method will handle the oauth flow by performing the following steps:
+   * - Opening a new window
+   * - Subscribe to message channel
+   * - Receive the token when ready
+   * - Close the opened tab
+   */
+  async login(onCloudLoginSuccess) {
+    const link = this._authLink();
+
+    const authWindow = window.open(link, '_blank', 'width=1024,height=716');
+
+    const handleToken = async e => {
+      // TODO: add security step to validate which domain the message is coming from
+      if (authWindow) {
+        authWindow.close();
+      }
+
+      window.removeEventListener('message', handleToken);
+
+      if (!e.data.token) {
+        Console.warn('Failed to login to Dropbox');
+        return;
+      }
+
+      this._dropbox.setAccessToken(e.data.token);
+      // save user name
+      const user = await this._getUser();
+
+      if (window.localStorage) {
+        window.localStorage.setItem(
+          'dropbox',
+          JSON.stringify({
+            // dropbox token doesn't expire unless revoked by the user
+            token: e.data.token,
+            user,
+            timestamp: new Date()
+          })
+        );
+      }
+
+      if (typeof onCloudLoginSuccess === 'function') {
+        onCloudLoginSuccess();
+      }
+    };
+
+    window.addEventListener('message', handleToken);
+  }
+
+  async downloadMap(loadParams) {
+
+    const token = this.getAccessToken();
+    if (!token) {
+      this.login(() => this.downloadMap(loadParams));
+    }
+    const result = await this._dropbox.filesDownload(loadParams);
+    const json = await this._readFile(result.fileBlob);
+
+    const response = {
+      map: json,
+      format: 'keplergl'
+    };
+
+    this._loadParam = loadParams;
+    return response;
+  }
+
+  async listMaps() {
+    // list files
+    try {
+      // https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesListFolder__anchor
+      const response = await this._dropbox.filesListFolder({
+        path: this._path
+      });
+      const {pngs, visualizations} = this._parseEntries(response);
+      // https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesGetThumbnailBatch__anchor
+      // up to 25 per request
+      // TODO: implement pagination, so we don't need to get all the thumbs all at once
+      const thumbnails = await Promise.all(this._getThumbnailRequests(pngs)).then(results =>
+        results.reduce((accu, r) => [...accu, ...(r.entries || [])], [])
+      );
+
+      // append to visualizations
+      thumbnails &&
+        thumbnails.forEach(thb => {
+          if (thb['.tag'] === 'success' && thb.thumbnail) {
+            const matchViz = visualizations[pngs[thb.metadata.id] && pngs[thb.metadata.id].name];
+            if (matchViz) {
+              matchViz.thumbnail = `data:image/gif;base64,${thb.thumbnail}`;
```

Comment:
let's create a constant for this prefix: ```data:image/gif;base64,```

Reply:
done